### PR TITLE
feat[UpaVerifier]: new verifyMixedAggregatedProof method

### PR DESCRIPTION
**Pull request by re1ro**
* [x] `UpaVerifier`: submitting `off-chain -> on-chain -> dummy` proofs into `verifyMixedAggregatedProof`

### CODE REVIEW

File: upa/contracts/UpaVerifier.sol [hunk 1]
```diff
@@ -292,7 +292,7 @@ contract UpaVerifier is
         SubmissionProof calldata submissionVerification,
         bytes32[] calldata proofIds,
         uint64 nextSubmissionIdx,
-        uint16 numOnChainProofs,
+        uint16 numProofs,
```
Comment (akileshtangella) [inline upa/contracts/UpaVerifier.sol]: I would leave this variable  name as `numOnChainProofs`. It makes it clearer that it refers to the number of on-chain proofs in the entire`verifyAggregatedProof` transaction, as opposed to the number of proofs in the submission (this quantity is also important, and is referred to as `numProofsInSubmission`) in the function.

**Review by akileshtangella — CHANGES_REQUESTED** [review]

File: upa/contracts/UpaVerifier.sol (hunk 1 above)
Comment (akileshtangella) [reply upa/contracts/UpaVerifier.sol]: Discussed w/ @re1ro to rename to `lastOnChainProofIdIdx`.

Comment (re1ro) [reply upa/contracts/UpaVerifier.sol]: Refactored to `lastOnChainProofIdIdx` as discussed

**Review by akileshtangella — APPROVED** [review]

File: upa/contracts/UpaVerifier.sol [hunk 2]
```diff
@@ -618,6 +622,161 @@ contract UpaVerifier is
...
+                // submission was verified over the course of multiple
+                // `verifyAggregatedProof` calls, only the last call would
+                // revert, leaving `currentOffChainSubmissionProofIds` stuck
+                // in an intermediate state.
+                if (verifierStorage.verifiedAtBlock[submissionId] == 0) {
+                    verifierStorage.verifiedAtBlock[submissionId] = block
+                        .number;
+
+                    emit SubmissionVerified(submissionId);
+                }
```
Comment (dtebbs) [inline upa/contracts/UpaVerifier.sol:697]: @akileshtangella Just flagging this so you are aware in the proof explorer.  We don't necessarily get an event if the submission was already verified.  We still have to compute the submission id and mark that in the DB, even if the corresponding event is not present.

**Review by dtebbs — APPROVED** [review]
LGTM

File: upa/contracts/UpaVerifier.sol (hunk 2 above)
Comment (re1ro) [reply upa/contracts/UpaVerifier.sol:697]: @akileshtangella this assumption will break our logic where we check for events to mark submission verified in DB.
Any thoughts?

**Pull request merged**